src/Documenter.jl

```diff
@@ -624,12 +624,14 @@ function git_push(
 
     try
         # Use a custom SSH config file to avoid overwriting the default user config.
+        ssh_dir = joinpath(homedir(), ".ssh")
+        isdir(ssh_dir) || mkdir(ssh_dir)
```

Comment:
This should be cleaned up if it was created.

Reply:
I moved this logic into `withfile` to create and clean up the directory